packages/application-generic/src/services/cache/cache.service.ts

```diff
@@ -152,37 +241,56 @@ export class CacheService implements ICacheService {
     }
   }
 
-  public delByPattern(pattern: string) {
-    const client = this.client;
-
-    if (client) {
-      return new Promise((resolve, reject) => {
-        const stream = this.inMemoryProviderService.inMemoryScan(pattern);
-
-        stream.on('data', function (keys) {
-          if (keys.length) {
-            const pipeline = client.pipeline();
-            keys.forEach(function (key) {
-              pipeline.del(key);
-            });
-            pipeline.exec().then(resolve).catch(reject);
-          }
-        });
-        stream.on('end', () => {
-          resolve(undefined);
-        });
-        stream.on('error', (err) => {
-          reject(err);
+  public async delByPattern(pattern: string) {
+    return this.usingClient(async (client) => {
+      if (client) {
+        return new Promise((resolve, reject) => {
+          const stream = this.inMemoryProviderService.inMemoryScan(pattern);
```

Comment:
Not sure about this part, but as i remember we don't/should not use delByPattern in the code anymore so maybe we can remove this method altogether.